3rdparty/CMakeLists.txt

```diff
@@ -2,6 +2,12 @@
 # Licensed under the MIT License.
 
 # Note: order does matter
+if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  # Enables all the warnings about constructions that some users consider questionable,
```

Comment:
I am not sure I understand this comment.

This should be saying: "While we enable `-Werror` globally in our CMake build, we do not want to fix all the warnings in third-party code. Thus we disable it here by adding `-Wno-error` for all targets under this folder."

Reply:
Also, this change appears separate from the change below. If we're going to disable treating warnings as errors for third-party code, shouldn't we then also remove all the code disabling specific warnings in third-party code?

Reply:
@yakman2020 @andschwa We do have our own code in 3rdparty (e.g. we customize libunwind that way; in future we would need to tweak libcxxrt as well to interface with the debugger when throwing exceptions). We want out code to be warnings free. So do you think -Wno-error in 3rdparty top-level is the right way?

Reply:
I think if we are heavily modifying code as you say, we should not have it in 3rdparty.